sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/RetryOptions.java

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.http.policy.RequestRetryCondition;
+import com.generic.core.models.HeaderName;
+import com.generic.core.util.ClientLogger;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+/**
+ * Options to configure the retry policy's behavior.
+ */
+public class RetryOptions {
+    private static final ClientLogger LOGGER = new ClientLogger(RetryOptions.class);
+    private int maxRetries;
+    private final Duration baseDelay;
+    private final Duration maxDelay;
+    private final Duration fixedDelay;
+    private Predicate<RequestRetryCondition> shouldRetryCondition;
+    private HashMap<HeaderName, Duration> retryHeaders = new HashMap<>();
```

Comment:
I think this would be better as a `Function<Headers, Duration>` as it would allow full control at call time of the function, allowing more advanced scenarios such a global backoff scenario (ex, we're seeing a lot of load issues right now, back everyone off more).

The default behavior should be the implementation method call that checks for the well-known retry headers.